### PR TITLE
Updating docker base image and useing docker entrypoint and new user

### PR DESCRIPTION
**Pull request by pselvana**
Removing security violations found in our docker image by upgrading to Ubuntu 23.10, adding a user and upgrading specific packages.

## Pull request type
Please check the type of change your PR introduces:

- Build related changes

## Changes
- upgrade Ubuntu 22.04 -> Ubuntu 23.10
- add datafoguser
- add new docker-entrypoint.sh to start app instead of trying to update the entrypoint constantly
- upgraded specific packages with known CVEs

Issue Number (if applicable): https://hub.docker.com/layers/datafog/datafog-api/1-latest/images/sha256-d2210762052aa18ca1c6d814124c7b412fd5a3bf4dc38154f40863d7217228b7?context=repo&tab=vulnerabilities

## Tests
% docker scout cves datafog-api:latest

## Other information

- Ubuntu 24.04 is blocked by DataFog-python not supporting Python 3.12 currently

### CODE REVIEW

**Comment by pselvana** [comment]
datafog-api git:(ci/docker-build) ✗ docker scout cves datafog-api:latest
    i New version 1.12.0 available (installed version is 1.11.0) at https://github.com/docker/scout-cli
    ✓ Image stored for indexing
    ✓ Indexed 308 packages
    ✗ Detected 2 vulnerable packages with a total of 2 vulnerabilities


## Overview

                    │       Analyzed Image         
────────────────────┼──────────────────────────────
  Target            │  datafog-api:latest          
    digest          │  78a4e0f26822                
    platform        │ linux/arm64/v8               
    vulnerabilities │    0C     1H     1M     0L   
    size            │ 362 MB                       
    packages        │ 308                          


## Packages and Vulnerabilities

   0C     1H     0M     0L  pip 24.2
pkg:pypi/pip@24.2

    ✗ HIGH CVE-2018-20225 [OWASP Top Ten 2017 Category A9 - Using Components with Known Vulnerabilities]
      https://scout.docker.com/v/CVE-2018-20225
      Affected range : >=0                                           
      Fixed version  : not fixed                                     
      CVSS Score     : 7.8                                           
      CVSS Vector    : CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H  
    

   0C     0H     1M     0L  requests 2.31.0
pkg:pypi/requests@2.31.0

    ✗ MEDIUM CVE-2024-35195 [Always-Incorrect Control Flow Implementation]
      https://scout.docker.com/v/CVE-2024-35195
      Affected range : <2.32.0                                       
      Fixed version  : 2.32.0                                        
      CVSS Score     : 5.6                                           
      CVSS Vector    : CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N  
    


2 vulnerabilities found in 2 packages
  LOW       0  
  MEDIUM    1  
  HIGH      1  
  CRITICAL  0

File: .github/workflows/main-docker.yml [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+
 on:
   workflow_run:
     workflows: ["Main and Dev CICD datafog-api app"]
     branches:
       - main
     types:
       - completed
+  pull_request:
+    branches: ["dev"]
```
Comment (sroy9675) [inline .github/workflows/main-docker.yml]: PR to main?

File: .github/workflows/main-docker.yml [hunk 2]
```diff
@@ -30,13 +36,23 @@ jobs:
...
           context: .
           file: ./Dockerfile
           push: true
-          tags: datafog/datafog-api:latest,datafog/datafog-api:1-latest
+          provenance: mode=max
+          sbom: true
+          tags: |
+            ${{ steps.meta.outputs.tags }}
+            ${{ env.IMAGE_NAME }}:latest
+            ${{ env.IMAGE_NAME }}:${{ env.MAJOR_VERSION }}
```
Comment (sroy9675) [inline .github/workflows/main-docker.yml:53]: just double checking you intentionally removed the latest from the version based tag?

File: app/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#/bin/sh
+. /.venv/bin/activate && uvicorn main:app --host 0.0.0.0 --port 8000
```
Comment (sroy9675) [inline app/docker-entrypoint.sh:2]: is the path to the venv correct?  and haven't we already activated the virtual env by this point?

**Review by sroy9675 — COMMENTED** [review]
A couple of minor questions

**Comment by pselvana** [comment]
image

image

File: .github/workflows/main-docker.yml (hunk 1 above)
Comment (pselvana) [reply .github/workflows/main-docker.yml]: Will remove in one sec. It was needed for testing.

File: .github/workflows/main-docker.yml (hunk 2 above)
Comment (pselvana) [reply .github/workflows/main-docker.yml:53]: Correct. That was a mistake in the first part. This keeps the major version as the fixed value (implicitly latest for that one)

File: app/docker-entrypoint.sh (hunk 3 above)
Comment (pselvana) [reply app/docker-entrypoint.sh:2]: It's activated in the docker step but not the runtime.  It would have to be re-activated in the entrypoint otherwise.

**Review by sroy9675 — APPROVED** [review]
looks good

**Pull request merged**